### ISSUE TITLE
Theming options available in admin and user theming should share a component

### ISSUE THREAD

**Issue by HannesJo0139** [nextcloud/server#35491]
With great pleasure I recognized the new user specific theming options. (Well, I'm not certain how new they actually are..) However, I wonder why you eg. use a different color picker in user theming than in admin theming panel.

Eg when I want to set a certain RGB code as background color, I can do it in admin panel but user panel is not capable of doing so. Here you invented the wheel twice and created a new / different color picker instead of sharing a color picker component.

The other way around, when setting background in admin theming panel, it would be great to have the same advanced options offered in user theming. Eg. why do you offer 15 backgrounds out-of-the-box for user theming, but in admin theming I can only reset to that awful clouds background image.

**Comment by skjnldsv** [comment]
With 26 they share the same color picker component now.


> but in admin theming I can only reset to that awful clouds background image.

You can also provide you own custom background. Admin theming is usually used for companies or people trying to set the tone for all users. Providing example would be unnecessary in my opinion.

- Companies already have their own precise idea on what their instances should look like color and background wise.
- Unique-users instances doesn't care about theming since we added the user theming with 25
- Small personal instances with a few users usually do not care about enforcing the same design to their users.

But more generally speaking, they serve different goals.

**Comment by skjnldsv** [comment]
> Eg when I want to set a certain RGB code as background color, I can do it in admin panel but user panel is not capable of doing so

We could [enable the advanced settings](https://user-images.githubusercontent.com/14975046/205298960-7eee6f4a-97d8-4749-b88d-668353d3622c.png) for the colour picker. But users usually do not know about hex colours
Admin are most likely to.

**Comment by HannesJo0139** [comment]
Ok I get your point. But I don't think the strict targeting of companies in the admin context and "dumb average end users" in the user context is accurate.

I run a private instance of about 10 users. Your comments on admin settings don't apply to me. I am no company. Now, if I want to set the "Cityscape" image as the uniform background for all users, I have to copy the image out of nextcloud's files and upload it again via the admin context. At the same time, it would not be a disadvantage for companies to have the additional background options. Whether they use them or not.

Regarding the Color Picker: Let's take my girlfriend for instance. She is the typical end user without technical know how. She is for some reason totally into Pantone colors and just asked me if there is a way to set them as background color. I tell her, yes you need the RGB code of your color. You can enter it in the color picker... and here we go.

That being said. I see your point, but for users who are not average, I think it sets unnecessary restrictions. PLUS: If you equalized the options, you could save a lot of code.

**Comment by Starcr0ss** [comment]
I agree with this issue that OP posted. I am frustrated that I can change my accent color to any hex code in the admin context, but can only use the color picker and _not_ specify a hex color code in a regular, non admin context. 

I know that as a single user instance, it doesn't make too much of a difference, but I still would like to be able to set my accent color with a non-elevated user/using the per-user settings. The new color picker lacking a hex code box also weakens the power of apps like Calendar, limiting my ability to finely select my calendar color. 

I don't get why the color picker would be _dumbed down_ with an option to make it more advanced. Something like hiding the hex color text box under a dropdown would be more than enough.
